pkg/registry/apps/replicaset/storage/storage.go

```diff
@@ -200,10 +201,11 @@ func (r *ScaleREST) Update(ctx context.Context, name string, objInfo rest.Update
 
 	rs.Spec.Replicas = scale.Spec.Replicas
 	rs.ResourceVersion = scale.ResourceVersion
-	rs, err = r.registry.UpdateReplicaSet(ctx, rs, createValidation, updateValidation, options)
+	obj, _, err = r.store.Update(ctx, rs.Name, rest.DefaultUpdatedObjectInfo(rs), createValidation, updateValidation, false, options)
```

Comment:
@liggitt isn't here a race between multiple updating actors?

Assume:
- the request comes in with scale `resourceVersion=50` (maybe because I am faking the `resourceVersion`)
- L174: r.store.Get returns `resourceVersion=42`
- L203: overrides `rs.resourceVersion=50` which is actually based on `resourceVesion=42`
- meanwhile there were actual updates, reaching the real `resourceVersion=50`
- L204: update succeeds, wiping all changes out since `resourceVersion=42`.

Reply:
The incoming request faking the resource version and happening to coincide with the actual resourceVersion isn't a scenario we care about. The same issue would exist with any resource.

Reply:
A more interesting case is when the scale resource version is not set. Stepping back, I think updating the rs from the incoming scale object should happen inside an updatedObjectInfo implementation, and the rs resourceVersion be left as is if the incoming scale object didn't set it. See a similar approach described at https://github.com/kubernetes/kubernetes/pull/59416#discussion_r213531564